File: 0011-container-with-most-water/0011-container-with-most-water.go
func maxArea(height []int) int {
    l := 0; r := len(height) - 1; res := 0;
    for l < r {
        res = max(res,min(height[r],height[l]) * (r - l));
        if height[l] < height[r] {
            l += 1
        } else {
            r -= 1
        }
    }
    
    return res
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l, r = 0, len(height) - 1
        res = 0
        while l < r:
            tmp = min(height[l],height[r]) * (r - l)
            res = max(res, tmp)
            if height[l] < height[r]: 
                l += 1
            else: 
                r -= 1
        return res
        